plugins/org.eclipse.xtend.ide/src/org/eclipse/xtend/ide/autoedit/RichStringPartionIndentationStrategy.java

```diff
@@ -33,14 +36,12 @@ public void customizeDocumentCommand(IDocument d, DocumentCommand c) {
 					regex+= Pattern.quote(s);
 				}
 				regex += "]";
-				String[] lines = pastedText.split(regex);
+				String[] lines = c.text.split(regex);
 				String convertedText = lines[0];
 				for(int i = 1; i < lines.length ; i++){
-					String oneLine = lines[i];
-					convertedText += d.getLineDelimiter(line) +  lineIndentation + oneLine;
+					convertedText += ((IDocumentExtension4) d).getDefaultLineDelimiter() +  lineIndentation + lines[i];
```

Comment:
We should better use a stringbuilder here instead of applying text in a loop
Also we should read the lineDelimiter only once